### PR TITLE
Make WhoIs-related commands work

### PR DESCRIPTION
**Pull request by WhittlesJr**
As it turns out, everything we were doing related to sending WhoIs requests was not actually doing anything. Even if we sent WhoIs requests, the remote device cache for the local device would not change. 

As I investigated this, I found that the only way LocalDevice.remoteDeviceCache was being manipulated was through BACnet4J's automagical RemoteDeviceDiscoverer or RemoteDeviceFinder classes, which do their own WhoIs broadcasts and listen for IAm responses. 

The only reason that discover-network appeared to work was because, on init, we start RemoteDeviceDiscoverer, which does its own WhoIs and finds everyone for us and caches them. You could literally comment out the i-am-broadcast! and discover-network in bc-core/boot-up! and it would still work exactly the same. find-remote-device and all of its friends were basically well-written no-ops. (At least, as of BACnet4J 4.0.1)

So, bottom line, there was no way to send a custom WhoIs request, listen for IAm, and update the cache, without being inside the LocalDevice class. 

So, I added a hack that does what we need. Now you can actually do discover-network et. al. and they work as advertised. There's also a clear-remote-devices! method to play with.

### CODE REVIEW

**Comment by Frozenlock** [comment]
It is indeed probably a recent-ish change, as I recall having to manually send WhoIs.

I'm a little worried about getting the private field... I think I understood something about them not being available with java 9?
(See https://github.com/tobias/clojure-java-9/issues)

We could always ask the BACnet4 team to make the required methods available.

Thoughts?

**Comment by WhittlesJr** [comment]
Ah, excellent point. Yeah the private field access is very cringey and I would be very happy if the BACnet4J team could give us the hooks we need. (Better yet, it would be best if they could just provide us with a method to call a WhoIs and update the cache, without having to make our own listener. Then most of the code I wrote here could go away.)

**Comment by Frozenlock** [comment]
I was about to press the button to post the issue, but decided to revise a little further the changes in BACnet4J v4.

Many new methods are actually things we previously had to do manually. Unless we are losing some features, I think it's preferable to fallback on (wrap) BACnet4J's own functions.

For example, `RemoteDeviceFinder` is doing the same thing that our own `find-remote-device` was doing, but takes care of all the futures/caching/timeouts stuff proper to BACnet4J.
(Tho `find-remote-deviceS` might not have an equivalent...)

Do you have any specific uses for a manual WhoIs?

**Comment by WhittlesJr** [comment]
My use case is not typical, but my application is an automated test system. I have several test cases where we do whois over ranges including and not including the device's ID, and we test to ensure that our device either does or does not respond with an iAm. With this pull request I have everything I need. I could just internalize that code to my application to keep bacure in step with bacnet4j practices if you'd prefer... Although if I ever had to move to Java 9 that might still bite me.

**Comment by WhittlesJr** [comment]
I feel like bacnet4j could really benefit from exposing a method that allows for manual whois calls, to make it a more complete and holistic bacnet stack.

**Comment by Frozenlock** [comment]
> I feel like bacnet4j could really benefit from exposing a method that allows for manual whois calls, to make it a more complete and holistic bacnet stack.

You're right.
I'll try to open an issue tomorrow.

**Comment by WhittlesJr** [comment]
That would be wonderful, thank you!

**Comment by WhittlesJr** [comment]
As I started working on Who-Has/I-Have, I realized that all of these event listening services will be using the same DeviceEventAdapter and will have to update some kind of state managed by us anyway. So I decided that I would ditch trying to update caches inside of BACnet4J and instead just manage a separate cache as an atom (much like read-properties/aaa, although for a different class of events). 

So I'm closing this PR for now. I'll soon submit another one with Who-Has/I-Have added, and a locally managed event cache. No more hacking and much less reliance on BACnet4J playing nicely with us.

**Comment by Frozenlock** [comment]
That sounds very nice.
I'll push a very little modification in the meantime.
